services/provider/server/server.go

```diff
@@ -519,20 +534,84 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		Name: "noobaa-remote-join-secret",
 		Kind: "Secret",
 		Data: mustMarshal(map[string]string{
-			"auth_token": string(authToken),
-			"mgmt_addr":  noobaaMgmtAddress,
+			"auth_token":            string(authToken),
+			"mgmt_addr":             fmt.Sprintf("https://%s:443", noobaaMgmtAddress),
+			"AWS_ACCESS_KEY_ID":     string(accessKeyID),
+			"AWS_SECRET_ACCESS_KEY": string(secretAccessKey),
 		}),
 	})
 
 	extR = append(extR, &pb.ExternalResource{
-		Name: "noobaa-remote",
+		Name: "noobaa",
 		Kind: "Noobaa",
 		Data: mustMarshal(&nbv1.NooBaaSpec{
 			JoinSecret: &v1.SecretReference{
 				Name: "noobaa-remote-join-secret",
 			},
 		}),
 	})
+
+	nb := &nbv1.NooBaa{}
+	nb.Name = "noobaa"
+	nb.Namespace = s.namespace
+
+	if err = s.client.Get(ctx, client.ObjectKeyFromObject(nb), nb); err != nil {
+		return nil, fmt.Errorf("failed to get noobaa %v", err)
+	}
+	if nb.Status.Phase != nbv1.SystemPhaseReady {
+		// Noobaa system is not yet ready
+		return extR, nil
+	}
+	nbS3Endpoint := nb.Status.Services.ServiceS3.NodePorts[0]
+	endpointWithoutScheme := strings.TrimPrefix(nbS3Endpoint, "https://")
+	parts := strings.Split(endpointWithoutScheme, ":")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid S3 endpoint format: %s", nbS3Endpoint)
+	}
+
+	s3IP := parts[0]
+	s3Port, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return nil, fmt.Errorf("invalid S3 port format: %s", parts[1])
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: "s3-endpoint-proxy",
+		Kind: "ServiceSpec",
+		Data: mustMarshal(&v1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				{
+					Port:       443,
+					TargetPort: intstr.FromInt(s3Port),
+					Protocol:   corev1.ProtocolTCP,
+				},
+			},
+		}),
+	})
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: "s3-endpoint-proxy",
+		Kind: "Endpoints",
+		Data: mustMarshal(&v1.Endpoints{
+			Subsets: []v1.EndpointSubset{
+				{
+					Addresses: []v1.EndpointAddress{
```

Comment:
Is there any reason to use Endpoints rather than the newer recommended EndpointSlice?

Reply:
EndpointSlices introduce extra API objects, leading to more Kubernetes API traffic. Since Noobaa service only uses 1 backend pod using a single Endpoints object reduces unnecessary API operations.
Easier to debug and troubleshoot

Reply:
Fair enough, thanks.